### PR TITLE
Add fixes for map tool-tip and components

### PR DESCRIPTION
**Pull request by priyesh777**
## Addresses:
- Task: https://github.com/collective-service/ifrc-gates-dashboard/issues/186
- Task: https://github.com/collective-service/ifrc-gates-dashboard/issues/185
- Task: https://github.com/collective-service/ifrc-gates-dashboard/issues/117

## Depends:
- Server branch: https://github.com/collective-service/ifrc-gates-backend/pull/127#issue-1407094902

## Changes:
- Add fixes to map tool-tip and other components of map/
- Add changes to map sub-header component 

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] codegen errors
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

### CODE REVIEW

**Review by shreeyash07 — APPROVED** [review]

File: app/views/Dashboard/Overview/MapView/index.tsx [hunk 1]
```diff
@@ -183,7 +181,7 @@ interface ClickedPoint {
 
 interface TooltipProps {
     countryName: string | undefined;
-    indicatorValue: number | undefined;
+    indicatorValue: OverviewMapDataType | undefined;
```
Comment (subinasr) [inline app/views/Dashboard/Overview/MapView/index.tsx]: I don't think this is correct. Can you please verify?

File: app/views/Dashboard/Overview/MapView/index.tsx [hunk 2]
```diff
@@ -258,12 +288,12 @@ function Tooltip(props: TooltipProps) {
...
-                        <TextOutput
-                            description="(Outbreak)"
-                            value={normalizedForm(indicatorValue ?? 0)}
-                        />
-                    </>
+                    <NumberOutput
+                        suffix={indicatorValue?.format === 'percent' ? '%' : '(Outbreak)'}
+                        value={MapTooltipData(indicatorValue)}
+                        normal={indicatorValue?.format === 'raw'}
+                        precision="auto"
```
Comment (subinasr) [inline app/views/Dashboard/Overview/MapView/index.tsx]:
```suggestion
                        precision={1}
```

File: app/views/Dashboard/Overview/MapView/index.tsx [hunk 3]
```diff
@@ -383,6 +413,9 @@ function MapView(props: MapViewProps) {
 
     const recentHighValuesWithIndicator = mostRecentValues?.descMostRecentValues;
     const recentLowValuesWithIndicator = mostRecentValues?.ascMostRecentValues;
+    // Note: This sorting logic maybe required in future
```
Comment (subinasr) [inline app/views/Dashboard/Overview/MapView/index.tsx:407]:
```suggestion
    // NOTE: This sorting logic maybe required in future
```

File: app/views/Dashboard/Overview/index.tsx [hunk 4]
```diff
@@ -16,7 +16,24 @@ import { FilterType } from '../Filters';
...
+    if (filterValues?.indicator && filterValues?.outbreak) {
+        return `${selectedIndicatorName ?? filterValues?.indicator} for ${selectedOutbreakName ?? filterValues?.outbreak} - Latest available data`;
+    }
+    if (filterValues?.indicator) {
+        return `${selectedIndicatorName ?? filterValues?.indicator} - Latest available data`;
+    }
+    if (filterValues?.outbreak) {
+        return `Total number of cases by country for ${selectedOutbreakName ?? filterValues?.outbreak}`;
+    }
+    return 'Total number of cases by country';
```
Comment (subinasr) [inline app/views/Dashboard/Overview/index.tsx]:
```suggestion
    return 'New cases per million by country';
```

File: app/views/Dashboard/Overview/index.tsx [hunk 5]
```diff
@@ -25,7 +42,7 @@ interface Props {
     selectedOutbreakName: string | undefined;
 }
 
-function Overview(props: Props) {
+function Overview(props: OverViewProps) {
```
Comment (subinasr) [inline app/views/Dashboard/Overview/index.tsx]: Why??

File: app/views/Dashboard/Overview/index.tsx [hunk 6]
```diff
@@ -86,7 +103,11 @@ function Overview(props: Props) {
                         headingSize="small"
                         headingContainerClassName={styles.mapHeaderContainer}
                         headingDescription={currentTab === 'mapMode'
-                        && 'Total number of cases by country'}
+                            && (filterValues
+                                ? MapSubHeader(
+                                    filterValues, selectedIndicatorName, selectedOutbreakName,
+                                )
+                                : 'Total number of cases by country')}
```
Comment (subinasr) [inline app/views/Dashboard/Overview/index.tsx]: Doesn't MapSubHeader function handle the case of no filterValues?

**Review by subinasr — CHANGES_REQUESTED** [review]

**Review by subinasr — APPROVED** [review]

**Pull request merged**